src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs

```diff
@@ -160,6 +161,37 @@ private bool GetReadOnly(PropertyInfo property)
             return property.GetSetMethod() == null;
         }
 
+        private bool GetRequired(PropertyInfo property)
+        {
+            var publicCtor = property.DeclaringType?.GetConstructors().Where(c => c.IsPublic).FirstOrDefault();
+            if (publicCtor == null)
+            {
+                //ReferenceType for inheritance do not have public constructor, and currently there are ResourceData, TrackedResourceData, WritableResourceData.
+                //We treat Id, Name, ResourceType, Location in ResourceData/TrackedResourceData as required.
+                if (!property.DeclaringType!.Name.Equals("WritableResourceData", StringComparison.Ordinal))
+                {
+                    if (property.Name == "Id" ||
+                        property.Name == "Name" ||
+                        property.Name == "ResourceType" ||
+                        property.Name == "Location")
+                        return true;
+                }
```

Comment:
Although the swagger definition does not define id, name, type of ResourceData as required, but I believe in real cases they should be required and this avoids breaking changes in GAed libraries.